plugins/auth-backend/src/identity/KeyStores.ts

```diff
@@ -75,6 +75,15 @@ export class KeyStores {
       return keyStore;
     }
 
+    if (provider === 'static') {
+      const settings = ks?.getOptional<StaticKeyStoreConfig>('static');
```

Comment:
On one suggestion here too: try to avoid raw `get`/`getOptional` as much as possible. They throw away type safety and path tracking in the config system.

You may want to instead pass in the entire root config to `StaticKeyStore.fromConfig(config)` instead. Then that class can do stuff like `config.getConfigArray('auth.keystore.static.keys')` which

- ensures that that path exists at all
- that `keys` actually is an array
- that each entry in the array actually is an object

all in one fell swoop. Then you can go through that array and do `getString` and stuff on those entries, getting all of the type safety benefit. If something was wrong in the config, the user will helpfully be informed like `key foo.bar.baz was expected to be a string, but was a number` or so, helping them address the problem.

Reply:
Thanks for the tip! I keep struggling with correctly interpreting the config files. It always feels a bit messy. Hope this will help me. I was on holiday, still catching up. I'll try this out Monday to see how it works :D

Reply:
This should be fixed now and is now also better tested. Resolving this thread, but let me know if you want this done differently.